src/components/List/utils/ListContext.ts

```diff
@@ -0,0 +1,20 @@
+import React from "react";
+import { NOOP } from "../../../utils/function-utils";
+
+export let listIdCounter = 0;
+
+export const generateListId = () => `list-${listIdCounter++}`;
+
+type ListContextType = {
+  /**
+   * A callback function which is being called when the item is being focused: by keyboard navigation or by mouse hover
+   * @param ListItem index
+   */
+  updateFocusedItem: (index: number) => void;
```

Comment:
is this `index` or `id` that you get as argument?

Reply:
Changed it to `id`